common/src/mailbox_api.rs

```diff
@@ -112,6 +112,85 @@ impl Default for MailboxResp {
     }
 }
 
+#[cfg_attr(test, derive(PartialEq, Debug, Eq))]
+#[allow(clippy::large_enum_variant)]
+pub enum MailboxReq {
+    EcdsaVerify(EcdsaVerifyReq),
+    TestHmacVerify(HmacVerifyReq),
+    GetIdevCsr(MailboxReqHeader),
+    GetLdevCert(MailboxReqHeader),
+    StashMeasurement(StashMeasurementReq),
+    InvokeDpeCommand(InvokeDpeReq),
+    TestGetFmcAliasCert(MailboxReqHeader),
+    FipsVersion(MailboxReqHeader),
+    FwInfo(MailboxReqHeader),
+}
+
+impl MailboxReq {
+    pub fn as_bytes(&self) -> &[u8] {
+        match self {
+            MailboxReq::EcdsaVerify(req) => req.as_bytes(),
+            MailboxReq::TestHmacVerify(req) => req.as_bytes(),
+            MailboxReq::GetIdevCsr(req) => req.as_bytes(),
+            MailboxReq::GetLdevCert(req) => req.as_bytes(),
+            MailboxReq::StashMeasurement(req) => req.as_bytes(),
+            MailboxReq::InvokeDpeCommand(req) => req.as_bytes(),
+            MailboxReq::TestGetFmcAliasCert(req) => req.as_bytes(),
+            MailboxReq::FipsVersion(req) => req.as_bytes(),
+            MailboxReq::FwInfo(req) => req.as_bytes(),
+        }
+    }
+
+    pub fn as_bytes_mut(&mut self) -> &mut [u8] {
+        match self {
+            MailboxReq::EcdsaVerify(req) => req.as_bytes_mut(),
+            MailboxReq::TestHmacVerify(req) => req.as_bytes_mut(),
+            MailboxReq::GetIdevCsr(req) => req.as_bytes_mut(),
+            MailboxReq::GetLdevCert(req) => req.as_bytes_mut(),
+            MailboxReq::StashMeasurement(req) => req.as_bytes_mut(),
+            MailboxReq::InvokeDpeCommand(req) => req.as_bytes_mut(),
+            MailboxReq::TestGetFmcAliasCert(req) => req.as_bytes_mut(),
+            MailboxReq::FipsVersion(req) => req.as_bytes_mut(),
+            MailboxReq::FwInfo(req) => req.as_bytes_mut(),
+        }
+    }
+
+    pub fn cmd_code(&self) -> CommandId {
+        match self {
+            MailboxReq::EcdsaVerify(_) => CommandId::ECDSA384_VERIFY,
+            MailboxReq::TestHmacVerify(_) => CommandId::TEST_ONLY_HMAC384_VERIFY,
+            MailboxReq::GetIdevCsr(_) => CommandId::GET_IDEV_CSR,
+            MailboxReq::GetLdevCert(_) => CommandId::GET_LDEV_CERT,
+            MailboxReq::StashMeasurement(_) => CommandId::STASH_MEASUREMENT,
+            MailboxReq::InvokeDpeCommand(_) => CommandId::INVOKE_DPE,
+            MailboxReq::TestGetFmcAliasCert(_) => CommandId::TEST_ONLY_GET_FMC_ALIAS_CERT,
```

Comment:
Optional: should this be prepended with a test-only directive? (Ditto for above)